File: backend/HSE_Football_Backend/Models/Message.cs
using System.ComponentModel.DataAnnotations;

namespace HSE_Football_Backend.Models
{
    /// <summary>
    /// Сообщение
    /// </summary>
    public class Message
    {
        /// <summary>
        /// ID сообщения
        /// </summary>
        [Key]
        public long Id { get; set; }

        /// <summary>
        /// Время отправки
        /// </summary>
        public string? SendTime { get; set; }

        /// <summary>
        /// Текст
        /// </summary>
        [StringLength(301, ErrorMessage = "Слишком много символов")]
        public string? Text { get; set; }

        /// <summary>
        /// Прочитано ли
        /// </summary>
        public bool IsRead { get; set; }

        /// <summary>
        /// Номер телефона отправителя
        /// </summary>
        [RegularExpression(@"^([8]{1}[0-9]{10})?$", ErrorMessage = "Некорректный формат номера телефона")]
        public string? Sender { get; set; }

        /// <summary>
        /// Номер телефона получателя
        /// </summary>
        [RegularExpression(@"^([8]{1}[0-9]{10})?$", ErrorMessage = "Некорректный формат номера телефона")]
        public string? Receiver { get; set; }
    }
}

File: android/sources/features/chat/domain_chat/src/main/kotlin/andryuh/football/domain_chat/dto/Message.kt
package andryuh.football.domain_chat.dto

import android.os.Parcelable
import kotlinx.datetime.LocalDateTime
import kotlinx.parcelize.IgnoredOnParcel
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import kotlinx.serialization.decodeFromString
import kotlinx.serialization.json.Json

@Serializable
@Parcelize
data class Message(
  val id: String,
  @SerialName("sendTime") val sendTimeRaw: String? = null,
  @SerialName("text") val content: String,
  val isRead: Boolean,
  @SerialName("sender") val senderPhoneNumber: String,
  @SerialName("receiver") val receiverPhoneNumber: String,
) : Parcelable {

  @IgnoredOnParcel
  @Transient
  val sendTime: LocalDateTime? =
    runCatching<LocalDateTime?> { sendTimeRaw?.let(Json::decodeFromString) }.getOrNull()
}

@Serializable
data class SendMessageRequestBody(
  @SerialName("sender") val senderPhoneNumber: String,
  @SerialName("receiver") val receiverPhoneNumber: String,
  @SerialName("text") val content: String,
  val sendTime: @RawValue LocalDateTime,
)
